server/src/test/java/org/elasticsearch/index/query/RegexpQueryBuilderTests.java

```diff
@@ -46,6 +46,9 @@ protected RegexpQueryBuilder doCreateTestQueryBuilder() {
             }
             query.flags(flags.toArray(new RegexpFlag[flags.size()]));
         }
+        if (randomBoolean()) {
+            query.caseInsensitive(true);
```

Comment:
Can you also add a test that checks that setting this value to `false` throws an error ?